erpnext/public/js/controllers/buying.js

```diff
@@ -347,6 +347,12 @@ erpnext.buying.get_items_from_product_bundle = function(frm) {
 			freeze: true,
 			callback: function(r) {
 				if(!r.exc && r.message) {
+					if($.isArray(frm.doc.items) && frm.doc.items.length > 0) {
+						// remove first item row if empty
+						if(!frm.doc.items[0].item_code) {
```

Comment:
Item Code is not mandatory for Purchase Invoice, Invoice can be created with item_name, This will cause to delete the first row if item was added with item_name and other details